frontend/packages/ux-editor/src/components/config/editModal/EditOptions/OptionTabs/OptionTabs.tsx

```diff
@@ -34,89 +38,158 @@ export const OptionTabs = ({
+            <StudioTabs.Tab value={SelectedOptionsType.Manual}>
+              {t('ux_editor.options.tab_manual')}
+            </StudioTabs.Tab>
+            <StudioTabs.Tab value={SelectedOptionsType.ReferenceId}>
+              {t('ux_editor.options.tab_referenceId')}
+            </StudioTabs.Tab>
+          </StudioTabs.List>
+          <StudioTabs.Content className={classes.tabContent} value={SelectedOptionsType.CodeList}>
+            <EditOptionList component={component} handleComponentChange={handleComponentChange} />
+          </StudioTabs.Content>
+          <StudioTabs.Content value={SelectedOptionsType.Manual} className={classes.tabContent}>
+            <RenderManualOptionsV1
+              component={component}
+              handleComponentChange={handleComponentChange}
+              areLayoutOptionsSupported={renderOptions.areLayoutOptionsSupported}
+            />
+          </StudioTabs.Content>
+          <StudioTabs.Content
+            value={SelectedOptionsType.ReferenceId}
+            className={classes.tabContent}
+          >
+            <EditOptionListReference
+              component={component}
+              handleComponentChange={handleComponentChange}
+            />
+          </StudioTabs.Content>
+        </StudioTabs>
+      )}
+    </>
   );
 };
 
-type RenderManualOptionsProps = {
+type RenderOptionsProps = {
   areLayoutOptionsSupported: boolean;
 } & Pick<IGenericEditComponent<SelectionComponentType>, 'component' | 'handleComponentChange'>;
 
-const RenderManualOptions = ({
+const RenderOptions = ({
   component,
   handleComponentChange,
   areLayoutOptionsSupported,
-}: RenderManualOptionsProps) => {
+}: RenderOptionsProps) => {
   const errorMessage = useComponentErrorMessage(component);
   const { t } = useTranslation();
 
   if (areLayoutOptionsSupported === false) {
     return (
-      <StudioAlert className={classes.manualTabAlert} severity='info'>
+      <StudioAlert className={classes.tabAlert} severity='info'>
         {t('ux_editor.options.code_list_only')}
       </StudioAlert>
     );
   }
 
   return (
     <>
-      {shouldDisplayFeature('optionListEditor') ? (
-        <EditManualOptionsWithEditor
-          component={component}
-          handleComponentChange={handleComponentChange}
-        />
-      ) : (
-        <EditManualOptions component={component} handleComponentChange={handleComponentChange} />
+      <EditOptionChoice component={component} handleComponentChange={handleComponentChange} />
+      {errorMessage && component.options !== undefined && (
+        <StudioErrorMessage className={classes.errorMessage} size='small'>
+          {errorMessage}
+        </StudioErrorMessage>
       )}
+    </>
+  );
+};
+
+// Todo: Remove once featureFlag "optionListEditor" is removed.
+type RenderManualOptionsV1Props = {
+  areLayoutOptionsSupported: boolean;
+} & Pick<IGenericEditComponent<SelectionComponentType>, 'component' | 'handleComponentChange'>;
```

Comment:
I think we will avoid having these "almost duplicate" functions in the same files, if we do a hard separation of the old and new features as suggested in another comment 🤔